feat: add creatures and characters

TheDotOs opened the pull request:
This pull request includes several changes across multiple files to enhance the functionality of characters and creatures in the game. The most important changes include the addition of new methods for item collection and attack handling, the introduction of default attributes for specific characters, and the creation of default constructors for creatures with predefined attributes and behaviors.

### Enhancements to Character Functionality:

* [`core/src/main/java/io/github/com/ranie_borges/thejungle/model/entity/Character.java`](diffhunk://#diff-ee3282c89eeda37f807be36cd893f4b214e47a4ec3b5faca7e1a9dfedf815669L241-R242): Added the `collectItem` method to handle item collection logic.
* [`core/src/main/java/io/github/com/ranie_borges/thejungle/model/entity/characters/Doctor.java`](diffhunk://#diff-0f971288bfe7185bd457bd536bc467eb5c2dbc9d2e0f6a8f6b0f924e0cebb7fbL7-R75): Introduced attributes such as health, attack power, inventory, and methods to handle item collection, attack, and health recovery.
* [`core/src/main/java/io/github/com/ranie_borges/thejungle/model/entity/characters/Hunter.java`](diffhunk://#diff-06aaef3e03d85dfb39d3b95a7f164600abf93a4d41462ed931ff234a60d8bbc7L7-L28): Added default attributes and methods for item management and combat actions. [[1]](diffhunk://#diff-06aaef3e03d85dfb39d3b95a7f164600abf93a4d41462ed931ff234a60d8bbc7L7-L28) [[2]](diffhunk://#diff-06aaef3e03d85dfb39d3b95a7f164600abf93a4d41462ed931ff234a60d8bbc7L37-R48) [[3]](diffhunk://#diff-06aaef3e03d85dfb39d3b95a7f164600abf93a4d41462ed931ff234a60d8bbc7L46-R100)
* [`core/src/main/java/io/github/com/ranie_borges/thejungle/model/entity/characters/Lumberjack.java`](diffhunk://#diff-eb55e79fd79de19fcd211cb221d7d93c0a3d293d3b97526b3191f805df80d8e3L7-R75): Similar to Hunter, introduced default attributes and methods for item management and combat actions.
* [`core/src/main/java/io/github/com/ranie_borges/thejungle/model/entity/characters/Survivor.java`](diffhunk://#diff-34cb8cfbfd838a028657df4c32c7ee1e31c9e0c6885340105754238ddb5f540fL7-R74): Added default attributes and methods for item management and combat actions.

### Enhancements to Creature Functionality:

* [`core/src/main/java/io/github/com/ranie_borges/thejungle/model/entity/creatures/Bird.java`](diffhunk://#diff-6d9d0447e660b7f33de2e49426a7de6334b6b07278a51c8402aa04b1dd22c4e3R3-R11): Added a default constructor with predefined attributes and methods for creating drops and sprites. [[1]](diffhunk://#diff-6d9d0447e660b7f33de2e49426a7de6334b6b07278a51c8402aa04b1dd22c4e3R3-R11) [[2]](diffhunk://#diff-6d9d0447e660b7f33de2e49426a7de6334b6b07278a51c8402aa04b1dd22c4e3R30-R79)
* [`core/src/main/java/io/github/com/ranie_borges/thejungle/model/entity/creatures/Bunny.java`](diffhunk://#diff-f67674c8cf10555bb1830c25537f91b82ea139a9681589444383cee9944bd1e5R3-R11): Similar to Bird, added a default constructor with predefined attributes and methods for creating drops and sprites. [[1]](diffhunk://#diff-f67674c8cf10555bb1830c25537f91b82ea139a9681589444383cee9944bd1e5R3-R11) [[2]](diffhunk://#diff-f67674c8cf10555bb1830c25537f91b82ea139a9681589444383cee9944bd1e5R30-R79)
* [`core/src/main/java/io/github/com/ranie_borges/thejungle/model/entity/creatures/Cannibal.java`](diffhunk://#diff-d2ed60aa3e458ef14e18d1f517ba4032d97ba950bd7c4b99a8001a8568cf5edfR3-R11): Added a default constructor with predefined attributes and methods for creating sprites. [[1]](diffhunk://#diff-d2ed60aa3e458ef14e18d1f517ba4032d97ba950bd7c4b99a8001a8568cf5edfR3-R11) [[2]](diffhunk://#diff-d2ed60aa3e458ef14e18d1f517ba4032d97ba950bd7c4b99a8001a8568cf5edfR30-R51)
* [`core/src/main/java/io/github/com/ranie_borges/thejungle/model/entity/creatures/Deer.java`](diffhunk://#diff-878cfa7bc980af162582f1a1ce3fff461a5c80c71d99e80c8788cf5dbfcdd984R3-R11): Added imports for handling textures and sprites.

### Minor Fixes:

* [`core/src/main/java/io/github/com/ranie_borges/thejungle/model/entity/Item.java`](diffhunk://#diff-728518c093a0a02af81129534e25500d39bc42338604fefcce7e0eb01fb3befdL10-R10): Corrected the type of the `weight` parameter in the constructor.
* [`core/src/main/java/io/github/com/ranie_borges/thejungle/model/entity/NonPlayerCharacter.java`](diffhunk://#diff-8493f49f705a05531c89067ef37df8629237aeafdd33af795b29aa4f3683267dR151-R154): Added the `takeDamage` method to handle damage logic.

devgabrielsborges commented on core/src/main/java/io/github/com/ranie_borges/thejungle/model/entity/characters/Doctor.java:
```diff
@@ -4,70 +4,74 @@
+        if (item != null && inventory.contains(item)) {
+            System.out.println(getName() + " used: " + item.getName());
+            inventory.remove(item);
+            // Aqui você pode aplicar o efeito do item, se necessário
```
Não está usando o item de fato, apenas removendo do inventario

devgabrielsborges commented on core/src/main/java/io/github/com/ranie_borges/thejungle/model/entity/characters/Hunter.java:
```diff
@@ -4,28 +4,36 @@
+import java.util.List;
+
+//TODO implement default attributes for Character
 public class Hunter extends Character {
```
checar se esses atributos ja existem na classe Character

devgabrielsborges commented on core/src/main/java/io/github/com/ranie_borges/thejungle/model/entity/characters/Hunter.java:
```diff
@@ -43,31 +54,49 @@ public boolean attack(double attackDamage, NonPlayerCharacter npc) {
-
+        if (hasDrinkableItem) {
+            health = Math.min(health + 5, 100); // alterar a taxa de quanto cura
+            System.out.println(getName() + " drank and recovered health. Current health: " + health);
```
nao esta alterando a vida do personagem de fato, tem que usar o set

devgabrielsborges commented on core/src/main/java/io/github/com/ranie_borges/thejungle/model/entity/characters/Hunter.java:
```diff
@@ -43,31 +54,49 @@ public boolean attack(double attackDamage, NonPlayerCharacter npc) {
+        if (item != null && inventory.contains(item)) {
+            System.out.println(getName() + " used: " + item.getName());
+            inventory.remove(item);
+            // aplicar efeito do item se necessário
```
item nao eh usado, apenas removido do inventario

devgabrielsborges commented on core/src/main/java/io/github/com/ranie_borges/thejungle/model/entity/characters/Lumberjack.java:
```diff
@@ -4,70 +4,74 @@
+        if (item != null && inventory.contains(item)) {
+            System.out.println(getName() + " used: " + item.getName());
+            inventory.remove(item);
+            // aqui você pode aplicar o efeito do item se for relevante
```
item nao eh usado

devgabrielsborges commented on core/src/main/java/io/github/com/ranie_borges/thejungle/model/entity/creatures/Bunny.java:
```diff
@@ -23,8 +27,54 @@ protected Bunny(
+
+    private static Map<String, Sprite> createSprites() {
+        Map<String, Sprite> sprites = new HashMap<>();
+        sprites.put("idle", new Sprite(new Texture(Gdx.files.internal("imagemAqui"))));
```
cuidado para nao esquecer de botar o path da imagem depois, bom adicionar um todo

devgabrielsborges commented on core/src/main/java/io/github/com/ranie_borges/thejungle/model/entity/creatures/Bunny.java:
```diff
@@ -23,8 +27,54 @@ protected Bunny(
+    }
+
     @Override
     public void attack() {
```
nao esta atacando

TheDotOs commented:
This pull request introduces significant updates to the character and creature classes within the game, focusing on expanding functionality, improving behaviors, and adding new features. Key changes include the addition of new methods for handling interactions such as item collection, attacks, and drinking, as well as the implementation of default constructors and sprite/drop generation for creatures.

### Character Enhancements
* **New Methods for Interaction**: Replaced the `defend` method with `collectItem` in the `Character` class, enabling characters to pick up items with inventory checks. (`Character.java`, [core/src/main/java/io/github/com/ranie_borges/thejungle/model/entity/Character.javaL241-R242](diffhunk://#diff-ee3282c89eeda37f807be36cd893f4b214e47a4ec3b5faca7e1a9dfedf815669L241-R242))
* **Additional Behavior for Non-Player Characters**: Added a `takeDamage` method to `NonPlayerCharacter` for handling damage logic. (`NonPlayerCharacter.java`, [core/src/main/java/io/github/com/ranie_borges/thejungle/model/entity/NonPlayerCharacter.javaR151-R154](diffhunk://#diff-8493f49f705a05531c89067ef37df8629237aeafdd33af795b29aa4f3683267dR151-R154))

### Specific Character Implementations
* **Doctor, Hunter, Lumberjack, and Survivor Classes**: 
  - Added constructors with default attributes (e.g., life, attack damage, inventory capacity). 
  - Implemented methods for attacking creatures, avoiding fights, collecting items, drinking, and using items. (`Doctor.java`, [[1]](diffhunk://#diff-0f971288bfe7185bd457bd536bc467eb5c2dbc9d2e0f6a8f6b0f924e0cebb7fbR3-R72); `Hunter.java`, [[2]](diffhunk://#diff-06aaef3e03d85dfb39d3b95a7f164600abf93a4d41462ed931ff234a60d8bbc7R3-R72); `Lumberjack.java`, [[3]](diffhunk://#diff-eb55e79fd79de19fcd211cb221d7d93c0a3d293d3b97526b3191f805df80d8e3R3-R72); `Survivor.java`, [[4]](diffhunk://#diff-34cb8cfbfd838a028657df4c32c7ee1e31c9e0c6885340105754238ddb5f540fR4-R71)

### Creature Enhancements
* **New Default Constructors**: Added default constructors for `Bird`, `Bunny`, and `Cannibal` with predefined attributes such as name, description, life ratio, and damage. (`Bird.java`, [[1]](diffhunk://#diff-6d9d0447e660b7f33de2e49426a7de6334b6b07278a51c8402aa04b1dd22c4e3R30-R80); `Bunny.java`, [[2]](diffhunk://#diff-f67674c8cf10555bb1830c25537f91b82ea139a9681589444383cee9944bd1e5R30-R79); `Cannibal.java`, [[3]](diffhunk://#diff-d2ed60aa3e458ef14e18d1f517ba4032d97ba950bd7c4b99a8001a8568cf5edfR3-R11)
* **Sprite and Drop Generation**: Introduced methods to generate sprites and item drops for creatures, enhancing their visual and gameplay interactions. (`Bird.java`, [[1]](diffhunk://#diff-6d9d0447e660b7f33de2e49426a7de6334b6b07278a51c8402aa04b1dd22c4e3R30-R80); `Bunny.java`, [[2]](diffhunk://#diff-f67674c8cf10555bb1830c25537f91b82ea139a9681589444383cee9944bd1e5R30-R79); `Cannibal.java`, [[3]](diffhunk://#diff-d2ed60aa3e458ef14e18d1f517ba4032d97ba950bd7c4b99a8001a8568cf5edfR3-R11)

These changes collectively enhance the game's functionality and provide a more dynamic and interactive experience for players.

TheDotOs reviewed (commented):
pronto
